tests/integrations/realcluster/real_cluster.go

```diff
@@ -0,0 +1,167 @@
+	// If the cluster does not fail to deploy, the cluster will be destroyed in
+	// the cleanup function. And these code will not work.
+	s.clusterCnt++
+	s.stopRealCluster(s.T())
+}
+
+func (s *realClusterSuite) startRealCluster(t *testing.T) {
+	log.Info("start to deploy a real cluster")
+
+	s.deploy(t)
+	s.clusterCnt++
+}
+
+func (s *realClusterSuite) stopRealCluster(t *testing.T) {
+	s.clusterCnt--
+
+	log.Info("start to destroy a real cluster", zap.String("tag", s.tag()))
+	destroy(t, s.tag())
+	time.Sleep(5 * time.Second)
+}
+
+func (s *realClusterSuite) tag() string {
+	return fmt.Sprintf("pd_real_cluster_test_%s_%d", s.suiteName, s.clusterCnt)
+}
+
+// func restartTiUP() {
+// 	log.Info("start to restart TiUP")
+// 	cmd := exec.Command("make", "deploy")
+// 	cmd.Stdout = os.Stdout
+// 	cmd.Stderr = os.Stderr
+// 	err := cmd.Run()
+// 	if err != nil {
+// 		panic(err)
+// 	}
+// 	log.Info("TiUP restart success")
+// }
+
+func (s *realClusterSuite) deploy(t *testing.T) {
+	tag := s.tag()
+	deployTiupPlayground(t, tag)
+	waitTiupReady(t, tag)
+}
+
+func destroy(t *testing.T, tag string) {
+	cmdStr := fmt.Sprintf("ps -ef | grep 'tiup playground' | grep %s | awk '{print $2}' | head -n 1", tag)
+	cmd := exec.Command("sh", "-c", cmdStr)
+	bytes, err := cmd.Output()
+	require.NoError(t, err)
+	pid := string(bytes)
+	// nolint:errcheck
+	runCommand("sh", "-c", "kill -9 "+pid)
+	log.Info("destroy success", zap.String("pid", pid))
+}
+
+func deployTiupPlayground(t *testing.T, tag string) {
+	curPath, err := os.Getwd()
+	require.NoError(t, err)
+
+	log.Info(curPath)
+	require.NoError(t, os.Chdir("../../.."))
+
+	if !fileExists("third_bin") || !fileExists("third_bin/tikv-server") || !fileExists("third_bin/tidb-server") || !fileExists("third_bin/tiflash") {
+		log.Info("downloading binaries...")
+		log.Info("this may take a few minutes, you can also download them manually and put them in the bin directory.")
+		require.NoError(t, runCommand("sh",
+			"./tests/integrations/realcluster/download_integration_test_binaries.sh"))
+	}
+	if !fileExists("bin") || !fileExists("bin/pd-server") {
+		log.Info("complie pd binaries...")
+		require.NoError(t, runCommand("make", "pd-server"))
+	}
+	if !fileExists(filepath.Join(curPath, "playground")) {
+		require.NoError(t, os.Mkdir(filepath.Join(curPath, "playground"), 0755))
+	}
+	// nolint:errcheck
+	go runCommand("sh", "-c",
+		tiupBin+` playground nightly --kv 3 --tiflash 1 --db 1 --pd 3 \
+		--without-monitor --tag `+tag+` --pd.binpath ./bin/pd-server \
+		// --kv.binpath ./third_bin/tikv-server \
+		// --db.binpath ./third_bin/tidb-server --tiflash.binpath ./third_bin/tiflash \
+		--kv.binpath ./bin/tikv-server \
+		--db.binpath ./bin/tidb-server --tiflash.binpath ./bin/tiflash \
+		--pd.config ./tests/integrations/realcluster/pd.toml \
+		> `+filepath.Join(curPath, "playground", tag+".log")+` 2>&1 & `)
+
+	time.Sleep(10 * time.Second)
```

Comment:
What the purpose of it compared with waitTiupReady?

Reply:
To avoid to change the dir before execute `tiup playground`. I will add some comments for it.